### PR TITLE
Pass better information of operation failures

### PR DESCRIPTION
**Pull request by skhugh**
#### What this PR does / why we need it?
Changes return type of public operations from Boolean to Result<Unit> for better information.
This can help users to handle exceptions more precisely.

Also bumps libraries to latest versions.

#### What are the relevant tickets?

Fixes #

### Checklist

- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: examples/kanban/src/main/java/com/example/kanbanapp/KanbanBoardViewModel.kt [hunk 1]
```diff
@@ -24,7 +24,9 @@ class KanbanBoardViewModel(private val client: Client) : ViewModel() {
 
     init {
         viewModelScope.launch {
-            if (client.activateAsync().await() && client.attachAsync(document).await()) {
+            if (client.activateAsync().await().isSuccess &&
+                client.attachAsync(document).await().isFailure
```
Comment (7hong13) [inline examples/kanban/src/main/java/com/example/kanbanapp/KanbanBoardViewModel.kt:28]: I think it should be `client.attachAsync(document).await().isSuccess`

File: yorkie/src/main/kotlin/dev/yorkie/document/Document.kt [hunk 2]
```diff
@@ -129,7 +130,7 @@ public class Document(
     public fun updateAsync(
         message: String? = null,
         updater: suspend (root: JsonObject, presence: Presence) -> Unit,
-    ): Deferred<Boolean> {
+    ): Deferred<Result<Unit>> {
```
Comment (7hong13) [inline yorkie/src/main/kotlin/dev/yorkie/document/Document.kt:133]: typealias `OperationResult` can be applied here

**Review by 7hong13 — APPROVED** [review]

File: examples/kanban/src/main/java/com/example/kanbanapp/KanbanBoardViewModel.kt (hunk 1 above)
Comment (skhugh) [reply examples/kanban/src/main/java/com/example/kanbanapp/KanbanBoardViewModel.kt]: done [bb70e2e](https://github.com/yorkie-team/yorkie-android-sdk/pull/179/commits/bb70e2ed131470b26e9f69fce4560c936a7035ec)

File: yorkie/src/main/kotlin/dev/yorkie/document/Document.kt (hunk 2 above)
Comment (skhugh) [reply yorkie/src/main/kotlin/dev/yorkie/document/Document.kt]: done [bb70e2e](https://github.com/yorkie-team/yorkie-android-sdk/pull/179/commits/bb70e2ed131470b26e9f69fce4560c936a7035ec)

**Pull request merged**